frontend/src/component/common/Table/cells/FeatureOverviewCell/FeatureOverviewCell.tsx

```diff
@@ -158,23 +168,79 @@ const ArchivedFeatureName: FC<{
     );
 };
 
-const RestTags: FC<{ tags: string[]; onClick: (tag: string) => void }> = ({
-    tags,
-    onClick,
-}) => {
+interface ITagItemProps {
+    tag: TagSchema;
+    onClick: (tag: TagSchema) => void;
+}
+
+const TagItem: FC<ITagItemProps> = ({ tag, onClick }) => {
+    const isTagTypeColorEnabled = useUiFlag('tagTypeColor');
+    const tagFullText = `${tag.type}:${tag.value}`;
+
+    if (isTagTypeColorEnabled) {
+        const tagComponent = (
+            <Box onClick={() => onClick(tag)} sx={{ cursor: 'pointer' }}>
+                <Tag tag={tag} maxLength={30} />
+            </Box>
+        );
+
+        return (
+            <HtmlTooltip key={tagFullText} title={tagFullText} arrow>
+                <span>{tagComponent}</span>
+            </HtmlTooltip>
+        );
+    }
+
+    // For non-color tags, use the StyledTag approach
+    const isOverflowing = tagFullText.length > 30;
+    const displayText = isOverflowing
+        ? `${tagFullText.substring(0, 30)}...`
+        : tagFullText;
+
+    return (
+        <StyledTag
+            key={tagFullText}
+            label={displayText}
+            size='small'
+            onClick={() => onClick(tag)}
+            sx={{ cursor: 'pointer' }}
+            title={isOverflowing ? tagFullText : undefined}
+        />
+    );
+};
+
+const RestTags: FC<{
+    tags: TagSchema[];
+    onClick: (tag: string) => void;
+}> = ({ tags, onClick }) => {
+    const isTagTypeColorEnabled = useUiFlag('tagTypeColor');
+
     return (
         <HtmlTooltip
             title={tags.map((tag) => (
                 <Box
                     sx={{ cursor: 'pointer' }}
-                    onClick={() => onClick(tag)}
-                    key={tag}
+                    onClick={() => onClick(`${tag.type}:${tag.value}`)}
```

Comment:
we keep constructing this type:value in so many places. Maybe we need a first class function for that?